app/jobs/regular/notify_reviewable.rb

```diff
@@ -74,4 +89,18 @@ def notify(user_ids, count:, updates:)
     @contacted += user_ids
   end
 
+  def notify_users(users, updates)
+    users.find_each { |user| notify_user(user, updates) }
+    @contacted += users.pluck(:id)
+  end
+
+  def notify_user(user, updates)
+    data = {
+      reviewable_count: user.reviewable_count,
+      unseen_reviewable_count: user.unseen_reviewable_count
+    }
+    data[:updates] = updates if updates.present?
+
+    MessageBus.publish("/reviewable_counts", data, user_ids: [user.id])
```

Comment:
I'm a little worried about this because we're not publishing one message per user instead of one message to multiple users. Can you expand on the differences between the old and new implementation on why we now need to publish user specific messages?

Reply:
The difference is that we now have the `unseen_reviewable_count` attribute which is based on the `last_seen_reviewable_id` column of `users` and therefor varies per user. When `unseen_reviewable_count` is larger than 0, you get a blue bubble on your avatar. It's also combined with `all_unread_notifications` so if any one of them is larger than 0, the blue bubble appears.

Note that in the current implementation we do publish a message per-user, but only for category moderators:

https://github.com/discourse/discourse/blob/b9c1e63bd1983c9d61da9681e8a653f5d97a85a9/app/jobs/regular/notify_reviewable.rb#L51-L61

Reply:
If this is the case, I think we need to publish it on the `/reviewable_counts/#{user.id}` channel instead so that we don't put the messages for all users into a single channel and result in the messageBus backlog overflowing.

Reply:
I'm not very familiar with how message bus works, so please bear with me. Does having too many channels cause other problems? Or I guess another way to put this question is when does it make sense to use a global channel vs a channel per user?

Reply:
The way MessageBus is designed is that each channel by default has a backlog size of 1000. This means that it'll keep the last 1000 published messages in Redis so that clients are able to catch up on messages during polling. When we centralise all the messages into a single channel, we're pooling every user's messages into a single backlog. On a large site, a new reviewable might have to be published to 20 users. It'll only take 50 reviewables in a short span of time for us to fill up the backlog as we would have published 1000 messages. In contrast to a channel that is scoped to each user, we can accommodate 1000 reviewables messages being published regardless of the number of users we have to publish to.

Reply:
ok, moved it to per-user channels.